### PR TITLE
Add a component prop to the button, so it is possible to use an exist…

### PR DESCRIPTION
**Pull request by taylorcode**
It would be nice to use an existing button component instance for the menu button. For example, I have a button component that has built-in tracking and dynamic behavior (ex. it turns into a "button link" if you pass a href prop).

As mentioned in the documentation, if we pass a button as the child to menu button it will not receive the tabIndex and role props.

This isn't necessary for the other components (wrapper, menu, menu item) because there are not element-specific props that we want to transfer.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -261,6 +263,8 @@ All props are optional.
 
 **tag** { String }: The HTML tag for this element. Default: `'span'`.
 
+**component** { Object }: A component instance to use instead of the tag.
+
 *Any additional props (e.g. `id`, `className`, `data-whatever`) are passed directly to the HTML element.*
```
Comment (davidtheclark) [inline README.md]: Let's change this line, too: `passed directly to the HTML element or custom component`.

**Comment by taylorcode** [comment]
Thanks David. I had to first update createMockManager so the tests would pass, and then I added a new test and updated the README.

File: lib/Button.js [hunk 2]
```diff
@@ -17,7 +18,10 @@ module.exports = React.createClass({
   },
 
   getDefaultProps: function() {
-    return { tag: 'span' };
+    return {
+      disabled: false,
```
Comment (davidtheclark) [inline lib/Button.js]: I'm fine with adding disabled here, though it's unrelated to the PR, right? 

If you do that, you should change its propType to `isRequired`, because it's always there.

File: lib/__tests__/helpers/createMockManager.js [hunk 3]
```diff
@@ -7,6 +7,7 @@ module.exports = function() {
     openMenu: jest.fn(),
     handleKeyDown: jest.fn(),
     handleClick: jest.fn(),
+    handleBlur: jest.fn(),
```
Comment (davidtheclark) [inline lib/__tests__/helpers/createMockManager.js]: Why have you added this? It's fine, but seems unrelated to the PR.

Comment (timweightman) [inline lib/__tests__/helpers/createMockManager.js]: preface: I'm new here-

I suspect that this is why `npm test` fails.

Snapshots expected `{undefined}`, and in reality the `onBlur` prop was not being added at all, and so the test would fail. It seems reasonable to either add a mock for `handleBlur` and expect `[Function]` in the snapshots, or else remove the `onBlur={undefined}` from each of the snapshots.

Comment (taylorcode) [inline lib/__tests__/helpers/createMockManager.js]: That's right. I mentioned above that I needed to update createMockManager because tests are failing and I cannot add a new test. I'll separate this into a new PR.

**Comment by taylorcode** [comment]
David when you mention "There are a couple of significant problems with using a non-standard DOM element as the button component." can you give an example of what you mean? If you pass a component it ultimately renders a dom element, so I'm not sure what a non-standard dom element is.

The main motivation is I have a button I would like to use as the menu's button. It has click-tracking and some props to configure appearance and behavior. I would prefer not to recreate the functionality of this button using the aria Button.

I could pass my button as children, but the documentation mentions: "The Button component itself acts as a UI button (with tab-index, role, etc.), so you probably do not want to pass an HTML <button> element as its child."

File: lib/Button.js (hunk 2 above)
Comment (taylorcode) [inline lib/Button.js]: I'll can move this to a new PR and add isRequired

File: lib/__tests__/helpers/createMockManager.js (hunk 3 above)
Comment (davidtheclark) [inline lib/__tests__/helpers/createMockManager.js]: You can just leave it in here. Strange that these tests have not failed for me, or for CI, though.

File: README.md [hunk 4]
```diff
@@ -261,7 +263,9 @@ All props are optional.
 
 **tag** { String }: The HTML tag for this element. Default: `'span'`.
 
-*Any additional props (e.g. `id`, `className`, `data-whatever`) are passed directly to the HTML element.*
+**component** { Object }: A component instance to use instead of the tag.
+
```
Comment (davidtheclark) [inline README.md]: I think an example is necessary, along with an explanation of why you might want to do this.

File: lib/__tests__/Button.test.js [hunk 5]
```diff
@@ -52,6 +52,21 @@ describe('<Button>', function() {
...
   });
 
+  it('DOM with some props and component instance used instead of tag', function() {
+    var button = el(
+      Button,
+      {
+        id: 'foo',
+        className: 'bar',
+        tag: 'div',
+        component: el('button'),
```
Comment (davidtheclark) [inline lib/__tests__/Button.test.js]: I would prefer that the API does not involve passing an Element in, but the Component itself. That is, instead of passing `<MySpecialComponent />` you pass `MySpecialComponent`. Does that make sense?

Comment (davidtheclark) [inline lib/__tests__/Button.test.js]: We need this test to verify behavior with a non-standard component. `button` can already be covered by the existing functionality: let's test the extent of the new functionality.

**Comment by taylorcode** [comment]
Hey again David,

After thinking it over, instead of adding this feature to your AriaButton, I can add it to my button component. If you prefer to keep the convention of creating the root element from a tag name, I don't mind closing this PR.

```
/*
  The main use case here:
  I'd like to use MyButton as the button to open a Menu, but also in other places.
  MyButton has styling, click tracking, and other features I want on every button.
  
  <MyButton /> should render a plain <button>.
  <MyAriaButton /> should render a menu-accessible button I can use to toggle a menu.
*/

class MyButton extends React.Component {
  render() {
    /* 
      this will override the onClick in the AriaButton,
      if we do this the menu won't toggle.
    */
    const {component} = this.props
    const rootComponent = component ? component : React.DOM.button;
    return <rootComponent onClick={trackClick} />;
  }
}

class MyAriaButton extends React.Component {
  render() {
    return <MyButton component={AriaButton} />;
  }
}
```

One problem I foresee with this approach is my onClick will override `handleClick` within the aria button.

Independent of this use-case it might be a nice to allow an onClick prop that will not override the menu-toggling.

Looking forward to your feedback on this.

**Comment by davidtheclark** [comment]
@taylorcode I think that after #56 you should be able to do what you need to do in a pretty straightforward way. Thanks for all your input!

**Comment by taylorcode** [comment]
Great, thank you David!
